upstream: http://public.kitware.com/mantis/view.php?id=9267
status: pending
origin: http://sisyphus.ru/ru/srpm/Sisyphus/cmake/patches/0
comment: fix cmake building when FORTIFY_SOURCE is enabled
--- a/Source/kwsys/SystemInformation.cxx
+++ b/Source/kwsys/SystemInformation.cxx
@@ -1896,9 +1896,14 @@
             case 5: sprintf (this->ChipID.ProcessorName,"Pentium II (0.25 micron)"); break;
             case 6: sprintf (this->ChipID.ProcessorName,"Pentium II With On-Die L2 Cache"); break;
             case 7: sprintf (this->ChipID.ProcessorName,"Pentium III (0.25 micron)"); break;
+/*
             case 8: sprintf (this->ChipID.ProcessorName,"Pentium III (0.18 micron) With 256 KB On-Die L2 Cache "); break;
             case 0xa: sprintf (this->ChipID.ProcessorName,"Pentium III (0.18 micron) With 1 Or 2 MB On-Die L2 Cache "); break;
             case 0xb: sprintf (this->ChipID.ProcessorName,"Pentium III (0.13 micron) With 256 Or 512 KB On-Die L2 Cache "); break;
+*/
+            case 8: sprintf (this->ChipID.ProcessorName,"Pentium III (0.18 micron) With 256 KB L2 Cache "); break;
+            case 0xa: sprintf (this->ChipID.ProcessorName,"Pentium III (0.18 micron), 1/2 MB L2 Cache "); break;
+            case 0xb: sprintf (this->ChipID.ProcessorName,"Pentium III (0.13 micron), 256/512 KB L2 Cache "); break;
             default: sprintf (this->ChipID.ProcessorName,"Unknown P6 family"); return false;
             }
           break;
--- a/Utilities/cmtar/encode.c
+++ b/Utilities/cmtar/encode.c
@@ -32,7 +32,7 @@ th_finish(TAR *t)
   int i, sum = 0;
 
   if (t->options & TAR_GNU)
-    strncpy(t->th_buf.magic, "ustar  ", 8);
+    memcpy(t->th_buf.magic, "ustar  ", 8);
   else
   {
     strncpy(t->th_buf.version, TVERSION, TVERSLEN);
